src/Adapter/Attribute/Repository/AttributeRepository.php

```diff
@@ -69,6 +63,67 @@ public function __construct(
         $this->dbPrefix = $dbPrefix;
     }
 
+    /**
+     * @param AttributeId $attributeId
+     *
+     * @return ProductAttribute
+     *
+     * @throws AttributeNotFoundException
+     * @throws CoreException
+     */
+    public function get(AttributeId $attributeId): ProductAttribute
+    {
+        /** @var ProductAttribute $attribute */
+        $attribute = $this->getObjectModel(
+            $attributeId->getValue(),
+            ProductAttribute::class,
+            AttributeNotFoundException::class
+        );
+
+        return $attribute;
+    }
+
+    /**
+     * @param ProductAttribute $attribute
+     *
+     * @return AttributeId
+     *
+     * @throws CoreException
+     */
+    public function add(ProductAttribute $attribute): AttributeId
+    {
+        $attributeId = $this->addObjectModelToShops(
+            $attribute,
+            array_map(fn (int $shopId) => new ShopId((int) $shopId), $attribute->id_shop_list),
+            CannotAddAttributeException::class
+        );
+
+        return new AttributeId($attributeId);
+    }
+
+    /**
+     * @param ProductAttribute $attribute
+     * @param int $errorCode
+     *
+     * @return void
+     *
+     * @throws CoreException
+     */
+    public function update(ProductAttribute $attribute, int $errorCode = 0): void
```

Comment:
Is this one still used since the handlers rely on `partialUpdate` ? If the method is not used please remove it